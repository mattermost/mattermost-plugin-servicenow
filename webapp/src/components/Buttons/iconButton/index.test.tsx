// Copyright (c) 2022-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import IconButton from 'src/components/Buttons/iconButton';

describe('IconButton', () => {
    const baseProps = {
        tooltipText: 'mockTooltip',
    };

    it('should match snapshot with only tooltip text provided', () => {
        const wrapper = shallow(<IconButton {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot with all the props', () => {
        const props = {
            ...baseProps,
            iconClassName: 'mockIconClassName',
            extraClass: 'mockExtraClass',
            onClick: jest.fn(),
            children: (<></>),
        };
        const wrapper = shallow(<IconButton {...props}/>);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('Button').hasClass(props.extraClass)).toBeTruthy();
        expect(wrapper.find('i').hasClass(props.iconClassName)).toBeTruthy();
        wrapper.find('Button').simulate('click');
        expect(props.onClick).toHaveBeenCalled();
    });
});
