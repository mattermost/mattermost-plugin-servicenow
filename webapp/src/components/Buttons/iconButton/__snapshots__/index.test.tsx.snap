// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot with all the props 1`] = `
<Tooltip
  tooltipContent="mockTooltip"
>
  <Button
    active={false}
    aria-label="mockTooltip"
    className="plugin-btn servicenow-button-wrapper btn-icon mockExtraClass"
    disabled={false}
    onClick={[MockFunction]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="button"
    variant="outline-danger"
  >
    <i
      className="mockIconClassName"
    />
  </Button>
</Tooltip>
`;

exports[`IconButton should match snapshot with only tooltip text provided 1`] = `
<Tooltip
  tooltipContent="mockTooltip"
>
  <Button
    active={false}
    aria-label="mockTooltip"
    className="plugin-btn servicenow-button-wrapper btn-icon "
    disabled={false}
    onKeyDown={[Function]}
    tabIndex={0}
    type="button"
    variant="outline-danger"
  />
</Tooltip>
`;
